### DISCUSSION TITLE
Does Rdkit have a function to calculate the logS (Water solubility) of SMILES?

### DISCUSSION THREAD

**Discussion by miracle1111111** [rdkit/rdkit#4039] [Q&A]
I have two questions for your advice.
1、 I used it rdkit.Chem.QED To calculate the MW (molecular weight), algp (logP) and other attributes of smile. However, for the calculation of logs, I didn't find it in the document.
2、 For the calculation of formal charge, I have a doubt, such as the following code:
from rdkit import Chem
m = Chem.MolFromSmiles("CCCN(C1CC1)S(=O)(=O)C2=C(C)C=CC(=C2)C#N")
for atom in m.GetAtoms():
    print( atom.GetFormalCharge())#形式电荷 
![image](https://user-images.githubusercontent.com/49110370/114872221-b0fc3380-9e2c-11eb-9091-9417d2261bbe.png)

The result of the above output is to calculate the formal charge of each atom. Then I want to calculate the Formal Charge of the entire SMILES. Is it equivalent to the accumulation of the formal charge of each atom? thank you for your help!

**Comment by greglandrum** [comment]
1. The RDKit does not have a built-in logS model. I'm not aware of a published, high-quality model for water solubility which is suitable for integration with the core RDKit.
2. There's a function in the `Chem` module to calculate the formal charge of a molecule: `Chem.GetFormalCharge`

Here's example of that second one:
```
In [1]: from rdkit import Chem

In [2]: m = Chem.MolFromSmiles('[O-]CCC[NH3+]')

In [3]: Chem.GetFormalCharge(m)
Out[3]: 0
```

**Reply by miracle1111111** [reply]
Thank you for your help! For problems 1, my understanding is that logS (Water Solubility) can only be obtained by predicting and there is no suitable prediction model. Do you know any tools for calculating logS(Water Solubility) .

**Reply by sawsimeon** [reply]
Try ChemAxon, they do have the logS and pKa. This is kind of out of topic but I am interest to know how accurate is the standardization of molecules from the RDKit Standardization Pipeline. Is there a study that give us the idea on the accuracy of standardizer from ChemAxon, RDKit and KNIME. I mean if we try to strip salt from millions of molecules, how much percent of molecules will be left with salt from the output of each standardizer?
